lib/agent/actions/fileretrieval/index.js

```diff
@@ -13,107 +13,59 @@ var fs           = require('fs'),
     common       = require('./../../common'),
     needle       = require('needle'),
     join         = require('path').join,
-    api          = require('./../../plugins/control-panel/api'),
-    files        = require('./files_storage'),
-    EventEmitter = require('events').EventEmitter;
+    files        = require('./storage'),
+    Emitter      = require('events').EventEmitter;
 
-var system     = common.system,
-    config     = common.config,
-    protocol   = config._values['control-panel'].protocol,
-    host       = config._values['control-panel'].host,
-    url        = protocol + '://' + host,
-    node_bin   = path.join(system.paths.current, 'bin', 'node'),
-    logger     = common.logger,
-    os_name    = process.platform.replace('darwin', 'mac').replace('win32', 'windows');
+var system      = common.system,
+    run_as_user = common.system.run_as_user,
+    node_bin    = path.join(system.paths.current, 'bin', 'node'),
+    os_name     = process.platform.replace('darwin', 'mac').replace('win32', 'windows'),
+    logger      = common.logger;
 
 var em,
     cp;
 
-var UPLOAD_SERVER = url + '/upload/upload',
-    RESUMABLE_HEADER = 'X-Prey-Upload-Resumable',
-    OPEN_TIMEOUT  = 180000,
-    READ_TIMEOUT  = 5000;
-
+// check_pending_files is used to resume any files that might been pending. It's called from
+// filesagent/providers/network.
 exports.check_pending_files = function() {
   files.run_stored();
 }
 
 exports.start = function(options, cb) {
-  var file_path = options.path,
-      file_id = options.file_id,
-      file_size = parseInt(options.size),
-      user = options.user;
-
-  em = em || new EventEmitter();
-  if (cb) cb(null, em);
-
-  if (!options.resumable)
-    files.store(file_id, file_path, file_size, user);
-
-  var file = {
-    total: 0,
-    path: file_path,
-    user: user,
-    id: file_id,
-    size: file_size,
-    resumable: options.resumable
+  if (!options.resumable) {
+    options.resumable = false;
   }
 
-  retrieve_file(file, function() {
-    files.del(file.id);
-  });
-}
-
-function retrieve_file(file, cb) {
-  if (file.resumable) {
-    var url = UPLOAD_SERVER + '?uploadID=' + file.id;
-    // Make a call to get the last byte processed by the upload server
-    // before the disconnection in order to resume the upload from that position
-    needle.request('get', url, null, function(err, res) {
-      if (err) return em.emit(err);
-      var data = JSON.parse(res.body);
-      file.total = data.Total;
-      get_file(file, cb);
-    })
-    return;
-  }
-  get_file(file, cb);
-}
+  var argsv; 
+  if (os_name == 'windows')
```

Comment:
could you simplify this? `argsv` for a non windows os could work with `path.resolve`, for instance?